feat: Streamline State-Aware Markdown Editing with Toolbar-Based Actions

JiyaGupta-cs opened the pull request:
Closes #418 
This PR introduces a Markdown formatting toolbar to the TemplateMark editor to improve the authoring experience while working with templates. The toolbar provides quick-access formatting actions for commonly used Markdown constructs and integrates seamlessly with the existing editor behavior.

### Changes

1. Added a Markdown formatting toolbar with support for:

      - Headings (H1, H2, H3)
      - Bold
      - Italic
      - Links
      - Images
      - Ordered lists
      - Unordered lists

4. Implemented state-aware formatting behavior to prevent conflicting styles:

      - Applying H2 **automatically removes** H1 if already applied (and vice versa).
      - Bold / Italic toggles **unapply existing formatting** when reselected.
      - Ordered and unordered lists **switch cleanly** between each other instead of stacking.

5. Improved cursor placement for better UX:

      - When inserting a link or image, the cursor is automatically placed inside the () so users can immediately type the URL.
      - Selection and caret positioning are preserved to maintain writing flow.



### Screenshots or Video

[Screencast from 2025-12-16 01-37-27.webm](https://github.com/user-attachments/assets/d1b2ab24-c563-4cdc-b107-4c04fe63a269)


### Related Issues
- Issue #418 

### Author Checklist
- [x] Ensure you provide a [DCO sign-off](https://github.com/probot/dco#how-it-works) for your commits using the `--signoff` option of git commit.
- [ ] Vital features and changes captured in unit and/or integration tests
- [x] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [x] Extend the documentation, if necessary
- [x] Merging to `main` from `fork:branchname`

dselman commented on src/store/store.ts:
```diff
@@ -29,6 +29,17 @@ interface AppState {
   chatState: ChatState;
   aiConfig: AIConfig | null;
   chatAbortController: AbortController | null;
+  markdownEditorCommands?: {
```
I don't think we need these here - they are purely editor / view commands which update the template mark on save.

JiyaGupta-cs replied:
Agreed, thanks for the review.
I’ve moved the editor-specific command definitions out of ```store.ts``` and into the editor layer so that global state remains decoupled from view/editor concerns.

dselman commented on src/editors/editorsContainer/TemplateMarkdown.tsx:
```diff
@@ -7,22 +9,404 @@ function TemplateMarkdown() {
+    }
+  };
+
+  const applyLinePrefixToggle = (prefix: string, alternatives: string[] = []) => {
```
This seems verbose and repetitive. Can it be refactored and moved to make this easier to maintain?

JiyaGupta-cs commented:
@dselman I’m glad you liked the feature. Thank you for the review. I’ve addressed the feedback and updated the PR. Kindly review when you get a chance.

JiyaGupta-cs commented:
@dselman Thanks! I'll add that in the follow up PR
